src/engines/nwn2/area.cpp

```diff
@@ -559,6 +564,14 @@ void Area::loadCreatures(const Aurora::GFF3List &list) {
 	}
 }
 
+void Area::loadStores(const Aurora::GFF3List &list) {
+	for (Aurora::GFF3List::const_iterator s = list.begin(); s != list.end(); ++s) {
+		Store *store = new Store(**s);
```

Comment:
For what reason create `Store` on the heap? I think, this introduce memory leak. Stack allocation will be enough

Reply:
`loadObject()`: https://github.com/xoreos/xoreos/blob/master/src/engines/nwn2/area.cpp#L515

That method puts the object into the area's objects list, which then manages the memory of the object.
Yes, that it takes a reference was maybe not a good way to signal that it takes over the object.

Reply:
Since we're now C++11, that ObjectList should maybe be a `std::list<std::unique_ptr<Engines::NWN2::Object>>`? And loadObject then takes a `std::unique_ptr<Engines::NWN2::Object>` and `std::move()`s it into the list?

That's something for a future PR, though, I guess. And also something the other engines need to be changed to do as well.

Reply:
Do I need to create a unique_ptr then release it to the loadObject call?

Reply:
No. Right now, the code is fine as is. You're using loadObject as intended.

It could be just a bit clearer/nicer, but that's something for a new PR.

Reply:
Ok, then may be add comment about this in this PR, just to reduce confusing for newcomers?

Reply:
I added a comment.